doc: nrf70: debug: Add a section about collecting statistics (#13219)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf70/developing/debugging.rst

    cmake -GNinja -Bbuild -DBOARD=nrf7002dk_nrf5340_cpuapp -DSNIPPET=nrf70-debug samples/wifi/shell
    ninja -C build

Statistics
==========

> The nRF70 driver, firmware, and networking stack have statistics that can be enabled to help debug issues.

Reviewer: ```suggestion
The nRF70 driver, firmware, and networking stack have statistics feature that can be enabled to help debug issues.
```

You can enable statistics by either using ``nrf70-debug`` snippet, see `Enable debug features`_.

.. list-table:: Statistics table
    :header-rows: 1

    * - Command
      - Description
      - Functional area
    * - ``net stats``
      - Displays statistics for the networking stack, network interfaces, and network protocols.
      - Data path debugging (Networking stack)

>     * - ``wifi statistics``

Reviewer: Would like to confirm if it is ``wifi stats`` or ``wifi statistics``?
```suggestion
    * - ``wifi statistics``
```

Author: what is the diff here, both look same to me?

Reviewer: The intent was not to delete anything but just added a question. Please ignore this.

      - Displays frame statistics for the nRF70 driver.
      - Data path debugging (Wi-Fi driver)
    * - ``wifi_util tx_stats <vif_index>``
      - Displays transmit statistics for the nRF70 driver.
      - Data path debugging (Wi-Fi driver TX)
    * - ``wifi_util rpu_stats all`` [1]_
      - Displays statistics for the nRF70 firmware (all modules, support for specific modules is also available).
      - nRF70 Firmware debugging (data and control path)

> .. [1] This command only works when nRF70 control plane is functional, as it uses the control plane to retrieve the statistics.

Reviewer: ```suggestion
.. [1] This command only works when the nRF70 control plane is functional, as it uses the control plane to retrieve the statistics.
```

> .. note::
>     All statistics esp. data path statistics must be collected multiple times to see the incremental changes to understand the behavior.

Reviewer: ```suggestion
.. note::
   All statistics, especially data path statistics, must be collected multiple times to see the incremental changes and understand the behavior.
```
